lib/matplotlib/testing/decorators.py

```diff
@@ -393,11 +397,23 @@ def image_comparison(baseline_images, extensions=None, tol=0,
         Optional name for the base style to apply to the image test. The test
         itself can also apply additional styles if desired. Defaults to the
         '_classic_test' style.
-
     """
+
     if extensions is None:
         # default extensions to test
         extensions = ['png', 'pdf', 'svg']
+        if baseline_images is not None:
+            baseline_exts = [*filter(None, {Path(baseline).suffix[1:]
```

Comment:
What is `filter` doing for us here?  I think just the `{}` comprehension here and a list call below would be enough (or `list({})` here).

Reply:
It removes the empty (falsy) suffixes (so that passing `baseline_images=["foo"]` gets `baseline_exts = []`, not `=[""]`).

Reply:
fair enough.

Reply:
Can you please add a comment on the purpose of filter? It's really difficult to see what it's doing just from the code.